### PR TITLE
Add details to field report detail page

### PR DESCRIPTION
**Pull request by barshathakuri**
## This PR doesn't introduce:
- [x] typos
- [x] conflict markers
- [x] unwanted comments
- [x] temporary files, auto-generated files or secret keys
- [x] `console.log` meant for debugging
- [x] codegen errors

### CODE REVIEW

File: src/App/routes.tsx [hunk 1]
```diff
@@ -1101,6 +1101,36 @@ const fieldReportFormNew = customWrapRoute({
...
+    path: 'field-reports/:fieldReportId',
+    component: {
+        render: () => import('#views/FieldReportDetails'),
+        props: {},
+    },
+    parent: root,
+    wrapperComponent: Auth,
+    context: {
+        title: 'All Field Reports',
+        visibility: 'anything',
```
Comment (samshara) [inline src/App/routes.tsx]: Maybe we need to name the title suitably. Since it shows a specific field report.

File: src/App/routes.tsx [hunk 2]
```diff
@@ -1101,6 +1101,36 @@ const fieldReportFormNew = customWrapRoute({
...
+    },
+});
+
+//  FIX ME: Add field report form link
+
```
Comment (samshara) [inline src/App/routes.tsx]: Let's remove this. 
Note: it should be // FIXME:

File: src/views/FieldReportDetails/index.tsx [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+                    {strings.onLabel}
+                    <DateOutput
+                        value={fieldReportResponse?.created_at}
+                        format="dd-MM-yyyy"
+                    />
+                    (
+                    {fieldReportResponse?.regions?.map((region) => region.name)}
+                    -
+                    {fieldReportResponse?.districts?.map((district) => district.name)}
+                    )
```
Comment (samshara) [inline src/views/FieldReportDetails/index.tsx]: Isn't it better if we join these mapped region names with `,` .

File: src/views/FieldReportDetails/index.tsx [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+                    <>
+                        <Header
+                            heading={contact.ctype}
+                            headingLevel={4}
+                        />
+                        <div className={styles.information}>
+                            {contact.name}
+                        </div>
+                    </>
+                ))}
```
Comment (samshara) [inline src/views/FieldReportDetails/index.tsx]: Let's pass the key prop as we are using map to render.

File: src/App/routes.tsx (hunk 2 above)
Comment (barshathakuri) [reply src/App/routes.tsx]: Field report form in not available

File: src/App/routes.tsx [hunk 5]
```diff
@@ -1101,6 +1101,36 @@ const fieldReportFormNew = customWrapRoute({
...
+        visibility: 'anything',
+    },
+});
+
+// FIXME: Add field report form
+
+const fieldReportsFormEdit = customWrapRoute({
+    path: 'field-reports/:fieldReportId/edit',
+    component: {
+        render: () => import('#views/DrefApplicationForm'),
```
Comment (frozenhelium) [inline src/App/routes.tsx]: We already have a dummy view for it
```suggestion
        render: () => import('#views/FieldReportForm'),
```

File: src/App/routes.tsx [hunk 6]
```diff
@@ -1101,6 +1101,36 @@ const fieldReportFormNew = customWrapRoute({
...
+        props: {},
+    },
+    parent: root,
+    wrapperComponent: Auth,
+    context: {
+        title: 'Dref Application Form',
```
Comment (frozenhelium) [inline src/App/routes.tsx]:
```suggestion
        title: 'Field Report Form',
```

File: src/views/FieldReportDetails/i18n.json [hunk 7]
```diff
@@ -0,0 +1,42 @@
+{
+    "namespace": "fieldReportDetails",
+    "strings": {
+        "populationMovementSubtitle": "Population Movement |",
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/i18n.json]: This is the name of disaster type and should come from the field report

File: src/views/FieldReportDetails/index.tsx [hunk 8]
```diff
@@ -0,0 +1,275 @@
+import { Fragment } from 'react';
+
+import { useParams } from 'react-router-dom';
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]:
```suggestion
import { Fragment } from 'react';
import { useParams } from 'react-router-dom';
```

File: src/views/FieldReportDetails/index.tsx [hunk 9]
```diff
@@ -0,0 +1,275 @@
...
+import { useRequest } from '#utils/restRequest';
+
+import type { paths } from '#generated/types';
+import useTranslation from '#hooks/useTranslation';
+import Page from '#components/Page';
+import Link from '#components/Link';
+import Header from '#components/Header';
+import DateOutput from '#components/DateOutput';
+import TextOutput from '#components/TextOutput';
+
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]:
```suggestion

import useTranslation from '#hooks/useTranslation';
import Page from '#components/Page';
import Link from '#components/Link';
import Header from '#components/Header';
import DateOutput from '#components/DateOutput';
import TextOutput from '#components/TextOutput';
import { useRequest } from '#utils/restRequest';
import type { paths } from '#generated/types';
```

File: src/views/FieldReportDetails/index.tsx [hunk 10]
```diff
@@ -0,0 +1,275 @@
...
+        response: fieldReportResponse,
+    } = useRequest<FieldReportResponse>({
+        skip: isNotDefined(fieldReportId),
+        url: '/api/v2/field_report/{id}/',
+        pathVariables: {
+            id: fieldReportId,
+        },
+    });
+    const countries = fieldReportResponse?.countries;
+    const country = (countries ?? [])[0];
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: We can have multiple countries and need to show / handle the multiple countries

File: src/views/FieldReportDetails/index.tsx [hunk 11]
```diff
@@ -0,0 +1,275 @@
...
+
+    return (
+        <Page
+            mainSectionClassName={styles.fieldReportDetails}
+        >
+            <Header
+                heading={fieldReportResponse?.summary}
+                headingLevel={1}
+            />
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: We should use `heading` prop of `Page` component

File: src/views/FieldReportDetails/index.tsx [hunk 12]
```diff
@@ -0,0 +1,275 @@
...
+            <div className={styles.fieldReportSubtitle}>
+                {strings.populationMovementSubtitle}
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: We should show the disaster type here: `fieldReportResponse?.dtype?.name`

File: src/views/FieldReportDetails/index.tsx [hunk 13]
```diff
@@ -0,0 +1,275 @@
...
+                {country
+                    && (
+                        <Link
+                            className={styles.titleLink}
+                            to={`/countries/${country.id}#operations`}
+                        >
+                            {country.name}
+                        </Link>
+                    )}
+                |
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: 1. We should iterate over all the countries here and show them separated by comma.
2. We should use routes from `context/routes` and use `generatePath` to generate the URL

File: src/views/FieldReportDetails/index.tsx [hunk 14]
```diff
@@ -0,0 +1,275 @@
...
+                <Link
+                    className={styles.titleLink}
+                    to={`/emergencies/${fieldReportResponse?.event?.id}`}
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]:
```suggestion
                    to={generatePath(emergencyRoutes.absolutePath, { emergencyId: fieldReportResponse?.event?.id })}
```

File: src/views/FieldReportDetails/index.tsx [hunk 15]
```diff
@@ -0,0 +1,275 @@
...
+                    <DateOutput
+                        value={fieldReportResponse?.created_at}
+                        format="dd-MM-yyyy"
+                    />
+                    (
+                    {fieldReportResponse?.regions?.map((region) => region.name).join(',')}
+                    -
+                    {fieldReportResponse?.districts?.map((district) => district.name).join(',')}
+                    )
+                </div>
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: 1. This is not a key figure, so let's not add `keyFigure` classname 
2. We should use string template for this `Last updated by {user} on {date}` and resolve that here.

File: src/views/FieldReportDetails/index.tsx [hunk 16]
```diff
@@ -0,0 +1,275 @@
...
+                <Header
+                    heading={strings.numericDetailsTitle}
+                    headingLevel={3}
+                />
+                <div className={styles.fieldReportDetail}>
+                    <TextOutput
+                        labelClassName={styles.fieldReportLabel}
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: We should use container component for this

File: src/views/FieldReportDetails/index.tsx [hunk 17]
```diff
@@ -0,0 +1,275 @@
...
+                        label={strings.injuredRCLabel}
+                        value={fieldReportResponse?.num_injured}
+                    />
+                    <TextOutput
+                        labelClassName={styles.fieldReportLabel}
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: Let's not overwrite default styling of the text output label

File: src/views/FieldReportDetails/index.tsx [hunk 18]
```diff
@@ -0,0 +1,275 @@
...
+                    />
+                </div>
+                <Header
+                    heading={strings.descriptionTitle}
+                    headingLevel={3}
+                />
+                <div className={styles.descriptionDetails}>
+                    <TextOutput
+                        className={styles.descriptionValue}
+                        labelClassName={styles.descriptionLabel}
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: We should use `Container` here

File: src/views/FieldReportDetails/index.tsx [hunk 19]
```diff
@@ -0,0 +1,275 @@
...
+                    labelClassName={styles.assistanceLabel}
+                    label={strings.governmentRequestsInternationalAssistanceLabel}
+                    value={fieldReportResponse?.request_assistance}
+                />
+                <TextOutput
+                    className={styles.assistanceValue}
+                    labelClassName={styles.assistanceLabel}
+                    label={strings.nSRequestsInternationalAssistanceLabel}
+                    value={fieldReportResponse?.ns_request_assistance}
+                />
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: These should be grouped in a `Container`

File: src/views/FieldReportDetails/index.tsx [hunk 20]
```diff
@@ -0,0 +1,275 @@
...
+                <Header
+                    heading={strings.informationBulletinPublishedLabel}
+                    headingLevel={4}
+                />
+                <div className={styles.information}>
+                    {fieldReportResponse?.bulletin}
+                </div>
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: This should also be grouped in a `Container`

File: src/views/FieldReportDetails/index.tsx [hunk 21]
```diff
@@ -0,0 +1,275 @@
...
+                    <Fragment key={contact.id}>
+                        <Header
+                            heading={contact.ctype}
+                            headingLevel={4}
+                        />
+                        <div className={styles.information}>
+                            {contact.name}
+                        </div>
+                    </Fragment>
+                ))}
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: `Container` here as well

File: src/views/FieldReportDetails/index.tsx [hunk 22]
```diff
@@ -0,0 +1,275 @@
...
+
+    return (
+        <Page
+            mainSectionClassName={styles.fieldReportDetails}
+        >
+            <Header
+                heading={fieldReportResponse?.summary}
+                headingLevel={1}
+            />
+            <div className={styles.fieldReportSubtitle}>
```
Comment (frozenhelium) [inline src/views/FieldReportDetails/index.tsx]: We should use `description` prop of `Page` component for this

**Review by frozenhelium — CHANGES_REQUESTED** [review]
Lets also make sure that we don't leave any unused:
- className
- stylling
- translations

File: src/components/Header/styles.module.css [hunk 23]
```diff
@@ -10,6 +10,7 @@
 
         .heading-container {
             display: flex;
+            align-items: baseline;
```
Comment (tnagorra) [inline src/components/Header/styles.module.css]: Do we need to change the styling on the base component?

Comment (tnagorra) [reply src/components/Header/styles.module.css]: @frozenhelium

File: src/utils/common.ts [hunk 24]
```diff
@@ -360,3 +361,28 @@ export function getNumberOfDaysInMonth(year: number, month: number) {
     const dateWithLastDateOfPrevMonth = new Date(year, month + 1, 0);
     return dateWithLastDateOfPrevMonth.getDate();
 }
+
+export function joinList(list: React.ReactNode[], separator: React.ReactNode) {
```
Comment (tnagorra) [inline src/utils/common.ts:479]: is this moved from somewhere?

File: src/utils/constants.ts [hunk 25]
```diff
@@ -36,3 +36,8 @@ export const CATEGORY_RISK_LOW = 2;
 export const CATEGORY_RISK_MEDIUM = 3;
 export const CATEGORY_RISK_HIGH = 4;
 export const CATEGORY_RISK_VERY_HIGH = 5;
+
+export const FIELD_REPORT_STATUS_EARLY_WARNING = 8;
+export const FIELD_REPORT_STATUS_EVENT = 9;
+
+export const DISASTER_TYPE_EPIDEMIC = 1;
```
Comment (tnagorra) [inline src/utils/constants.ts]: Let's separate out the constants into two files?

1. general constants
2. go domain specific constants

@frozenhelium what do you think?

File: src/views/FieldReportDetails/index.tsx [hunk 26]
```diff
@@ -0,0 +1,391 @@
...
+                        withHeaderBorder
+                    >
+                        {fieldReportResponse?.external_partners.map(
+                            (partner) => (
+                                <div key={partner.id}>
+                                    {partner?.name}
+                                </div>
+                            ),
+                        )}
+                    </Container>
```
Comment (tnagorra) [inline src/views/FieldReportDetails/index.tsx]: Should we handle cases where external_partners is empty @frozenhelium ?

Either show an empty message or not show the container at all?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/views/FieldReportDetails/index.tsx (hunk 26 above)
Comment (frozenhelium) [reply src/views/FieldReportDetails/index.tsx]: Should hide the container

File: src/utils/constants.ts (hunk 25 above)
Comment (frozenhelium) [reply src/utils/constants.ts]: Can be done later

File: src/utils/common.ts (hunk 24 above)
Comment (frozenhelium) [reply src/utils/common.ts:479]: Nope, brand new stuff

File: src/components/Header/styles.module.css (hunk 23 above)
Comment (frozenhelium) [reply src/components/Header/styles.module.css]: Yess, I think so, it should already be the develop though

File: src/utils/common.ts (hunk 24 above)
Comment (tnagorra) [reply src/utils/common.ts:479]: Need to check if this stuff works :1234:

Comment (tnagorra) [reply src/utils/common.ts:479]: How are we defining keys here

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**